File: ddpoliglot_v6_flutter/lib/models/data/Log.dart
// ignore_for_file: file_names
import 'package:json_annotation/json_annotation.dart';

part 'Log.g.dart';

@JsonSerializable()
class Log {
  int logID;
  String userID;
  String name;
  String message;
  String description;
  int type;
  DateTime created;

  Log(
    this.logID,
    this.userID,
    this.name,
    this.message,
    this.description,
    this.type,
    this.created,
  );

  factory Log.fromJson(Map<String, dynamic> data) => _$LogFromJson(data);

  Map<String, dynamic> toJson() => _$LogToJson(this);
}

File: ddPoliglotV6.Data/Models/Log.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ddPoliglotV6.Data.Models
{
    public class Log
    {
        public int LogID { get; set; }
        
        public Guid UserID { get; set; }
        
        [MaxLength(1000)]
        public string Name { get; set; }

        [MaxLength(2000)]
        public string Message { get; set; }

        public string Description { get; set; }

        public int Type { get; set; }

        public DateTime Created { get; set; }
    }
}
